django/thunderstore/api/cyberstorm/views/community.py

```diff
@@ -0,0 +1,214 @@
+        communities_count = len(communities_qs)
+        serializer = self.serialize_results(
+            communities_page, communities_count, params["page_size"]
+        )
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def get_full_cache_vary(self, params: OrderedDict) -> str:
+        cache_vary = "communities_cache_key"
+        cache_vary += f".{params['ordering']}"
+        cache_vary += f".{params['page']}"
+        cache_vary += f".{params['page_size']}"
+        return cache_vary
+
+    def get_queryset(self, params=None) -> QuerySet[Community]:
+        return Community.objects.filter(is_listed=True,).annotate(
+            pkgs=Count(
+                "package_listings",
+                filter=Q(
+                    Q(package_listings__package__is_deprecated=False)
+                    & Q(
+                        Case(
+                            When(
+                                require_package_listing_approval=True,
+                                then=Case(
+                                    When(
+                                        Q(
+                                            package_listings__review_status=PackageListingReviewStatus.approved
+                                        ),
+                                        then=True,
+                                    ),
+                                    default=False,
+                                    output_field=BooleanField(),
+                                ),
+                            ),
+                            default=Case(
+                                When(
+                                    Q(
+                                        package_listings__review_status=PackageListingReviewStatus.unreviewed
+                                    )
+                                    | Q(
+                                        package_listings__review_status=PackageListingReviewStatus.approved
+                                    ),
+                                    then=True,
+                                ),
+                                default=False,
+                                output_field=BooleanField(),
+                            ),
+                        ),
+                    ),
+                ),
+                distinct=True,
+            ),
+            downloads=Sum(
+                Case(
+                    When(package_listings__package__is_deprecated=True, then=0),
+                    When(
+                        package_listings__review_status=PackageListingReviewStatus.rejected,
+                        then=0,
+                    ),
+                    default="package_listings__package__versions__downloads",
+                )
+            ),
+        )
+
+    def order_queryset(
+        self, ordering: str, queryset: QuerySet[Community]
+    ) -> QuerySet[Community]:
+        order_arg = {
```

Comment:
If we're going to order by name no matter what, just order by name without pretending otherwise?

Reply:
I've added that function there, kind of in advance. Wasn't sure with what do we need to order by.